### PR TITLE
Xinfra Monitor Rebranding Migration Continuation Part III

### PR DESCRIPTION
**Pull request by andrewchoi5**
Xinfra Monitor Rebranding Migration Continuation Part III
`Signed-off-by: Andrew Choi <li_andchoi@microsoft.com>`

### CODE REVIEW

File: src/main/java/com/linkedin/kmf/services/ProduceService.java [hunk 1]
```diff
@@ -276,8 +276,12 @@ public ProduceMetrics(final Metrics metrics, final Map<String, String> tags) {
...
           int partitionNum = _partitionNum.get();
           for (int partition = 0; partition < partitionNum; partition++) {
-            double recordsProduced = metrics.metrics().get(metrics.metricName("records-produced-rate-partition-" + partition, METRIC_GROUP_NAME, tags)).value();
-            double produceError = metrics.metrics().get(metrics.metricName("produce-error-rate-partition-" + partition, METRIC_GROUP_NAME, tags)).value();
+            double recordsProduced = (double) metrics.metrics()
+                .get(metrics.metricName("records-produced-rate-partition-" + partition, METRIC_GROUP_NAME, tags))
+                .metricValue();
+            double produceError = (double) metrics.metrics()
+                .get(metrics.metricName("produce-error-rate-partition-" + partition, METRIC_GROUP_NAME, tags))
+                .metricValue();
```
Comment (mgrubent) [inline src/main/java/com/linkedin/kmf/services/ProduceService.java:284]: I don't disagree with these changes, but they're not strictly part of rebranding

File: src/main/java/com/linkedin/kmf/services/ProduceService.java [hunk 2]
```diff
@@ -405,13 +409,13 @@ public void run() {
   private class ProduceServiceThreadFactory implements ThreadFactory {
 
     private final AtomicInteger _threadId = new AtomicInteger();
-    public Thread newThread(Runnable r) {
+    public Thread newThread(@SuppressWarnings("NullableProblems") Runnable r) {
```
Comment (mgrubent) [inline src/main/java/com/linkedin/kmf/services/ProduceService.java:412]: This seems outside of rebranding.  Would suggest that it gets split out into another commit for history hygiene.

File: src/main/java/com/linkedin/kmf/services/ProduceService.java [hunk 3]
```diff
@@ -405,13 +409,13 @@ public void run() {
...
       return new Thread(r, _name + "-produce-service-" + _threadId.getAndIncrement());
     }
   }
 
   private class HandleNewPartitionsThreadFactory implements ThreadFactory {
-    public Thread newThread(Runnable r) {
+    public Thread newThread(@SuppressWarnings("NullableProblems") Runnable r) {
```
Comment (mgrubent) [inline src/main/java/com/linkedin/kmf/services/ProduceService.java:418]: This seems outside of rebranding.  Would suggest that it gets split out into another commit for history hygiene.

File: src/main/java/com/linkedin/kmf/services/ProduceService.java [hunk 4]
```diff
@@ -64,7 +64,7 @@
   private final String _name;
   private final ProduceMetrics _sensors;
   private KMBaseProducer _producer;
-  private KMPartitioner _partitioner;
+  private final KMPartitioner _partitioner;
```
Comment (mgrubent) [inline src/main/java/com/linkedin/kmf/services/ProduceService.java:67]: This seems outside of rebranding.  Would suggest that it gets split out into another commit for history hygiene.

Comment (andrewchoi5) [reply src/main/java/com/linkedin/kmf/services/ProduceService.java:67]: sgtm

File: src/main/java/com/linkedin/kmf/services/ProduceService.java (hunk 3 above)
Comment (andrewchoi5) [reply src/main/java/com/linkedin/kmf/services/ProduceService.java:418]: sgtm

File: src/main/java/com/linkedin/kmf/services/ProduceService.java (hunk 2 above)
Comment (andrewchoi5) [reply src/main/java/com/linkedin/kmf/services/ProduceService.java:412]: sgtm

File: src/main/java/com/linkedin/kmf/services/ProduceService.java (hunk 1 above)
Comment (andrewchoi5) [reply src/main/java/com/linkedin/kmf/services/ProduceService.java:284]: sgtm

**Comment by andrewchoi5** [comment]
> Rebranding parts look fine.
> 
> Would prefer that changes in this series are limited to rebranding.

addressed. @mgrubent
